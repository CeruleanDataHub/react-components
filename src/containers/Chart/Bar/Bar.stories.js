import React from "react";

import { Bar } from "./Bar";

export default {
  title: "Bar",
  component: Bar
};

const xAxis = [
  {
    categories: [
      "monday",
      "tuesday",
      "wednesday",
      "thursday",
      "friday",
      "saturday",
      "sunday"
    ]
  }
];

const series = [
  { name: "Temperature", data: [99, 80, 82, 59, 90, 6, 99] },
  { name: "Humidity", data: [29, 39, 26, 53, 9, 46, 31] }
];

export const BarStory = () => (
  <Bar title="Sauna histogram as bar" xAxis={xAxis} series={series} />
);
