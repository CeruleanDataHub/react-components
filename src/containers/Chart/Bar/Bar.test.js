import { mount } from "enzyme";
import React from "react";

import { Bar } from "./Bar";

describe("Bar", () => {
  it("should render chart", () => {
    const component = mount(<Bar />);

    expect(component.render()).toMatchSnapshot();
  });

  it("should render chart with simple data", () => {
    const xAxis = [
      {
        categories: [
          "monday",
          "tuesday",
          "wednesday",
          "thursday",
          "friday",
          "saturday",
          "sunday"
        ]
      }
    ];

    const series = [
      { name: "temperature", data: [10, 15, 20, 25, 15, 20, 15] }
    ];

    const component = mount(
      <Bar title="Chart" xAxis={xAxis} series={series} />
    );

    expect(component.render()).toMatchSnapshot();
  });
});
