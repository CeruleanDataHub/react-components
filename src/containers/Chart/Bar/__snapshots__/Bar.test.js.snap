// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar should render chart 1`] = `
<Chart
  containerProps={null}
  options={Object {}}
  series={Array []}
  title=""
  type="bar"
  xAxis={null}
/>
`;

exports[`Bar should render chart with simple data 1`] = `
<Chart
  containerProps={null}
  options={Object {}}
  series={
    Array [
      Object {
        "data": Array [
          10,
          15,
          20,
          25,
          15,
          20,
          15,
        ],
        "name": "temperature",
      },
    ]
  }
  title="Chart"
  type="bar"
  xAxis={
    Array [
      Object {
        "categories": Array [
          "monday",
          "tuesday",
          "wednesday",
          "thursday",
          "friday",
          "saturday",
          "sunday",
        ],
      },
    ]
  }
/>
`;
